star-ros2/src/star_safety_monitor/include/star_safety_monitor/safety_monitor.hpp

```
@@ -0,0 +1,154 @@
+  // Timer callback for monitoring and publishing diagnostics
+  void monitoring_timer_callback();
+
+  // Monitoring and safety check methods
+  void check_heartbeat_health();
+  void check_velocity_limits();
+  void check_battery_health();
+  void check_motor_stall();
+  void check_diagnostic_health();
+  void update_overall_state();
+  void publish_diagnostics();
+  double compute_linear_magnitude(const geometry_msgs::msg::Twist & twist) const;
```

Clang (14.0.6)

[warning] function 'compute_linear_magnitude' should be marked [[nodiscard]]

(modernize-use-nodiscard)